dapps/shop/backend/utils/dns/clouddns.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * DNS utilities for GCP CLoud DNS
+ */
+
+const { DNS } = require('@google-cloud/dns')
+
+let CACHED_CLIENT
+const DEFAULT_TTL = 300 // 5 minutes
+
+/**
+ * Return a Google Cloud DNS API client
+ *
+ * @returns {DNS}
+ */
+function getClient(credentials) {
+  if (CACHED_CLIENT) return CACHED_CLIENT
+  if (!credentials) throw new Error('Must supply GCP credentails')
+  if (typeof credentials === 'string') credentials = JSON.parse(credentials)
+
+  CACHED_CLIENT = new DNS({ projectId: credentials.project_id, credentials })
+
+  return CACHED_CLIENT
+}
+
+/**
+ * Append e to s
+ *
+ * @param {string} s string we're appending to
+ * @param {string} e string we're appending
+ * @returns {string} s+e
+ */
+function append(s, e) {
+  if (s.endsWith(e)) return s
+  return `${s}${e}`
+}
+
+/**
+ * Looks for a record within a set of records matching name
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Record.html
+ *
+ * @param {Array} of Record
+ * @param {string} name of DNS record
+ * @returns {boolean} if a record matching name was found
+ */
+function containsRecord(records, name) {
```

Comment:
I think calls to this method can be replaced with ```records.includes(name)```.

Reply:
How's that?  `records` is an array of objects of which we need to check against the `name` property.

Reply:
doh. yes I read the code too fast!
I guess you could use the "some" function if you want ?
```records.some(r => r.name === name)```

Reply:
probably no gain in refactoring at this point but will keep it in mind for the future

Reply:
had to touch this code anyway, because of the getRecords result fix.  Refactored.